src/blockdev.rs

```diff
@@ -885,6 +885,10 @@ fn split_blkid_line(line: &str) -> HashMap<String, String> {
 }
 
 fn blkid() -> Result<Vec<HashMap<String, String>>> {
+    // Refresh the cache to avoid collecting old devices which no longer
+    // exist.
+    runcmd!("blkid", "-s", "none")?;
```

Comment:
How about we just change the `blkid -o device` call below to `blkid --cache-file /dev/null -o device` and add a comment? 

This would be if we didn't end up changing the strategy like Colin mentions in https://github.com/coreos/coreos-installer/pull/1288#issuecomment-1700989242

Reply:
That makes sense, I thought if we refresh the cache, then it would be better for later services which call blkid (to read from new catch), but agree it might have risk, I am OK with the change. Thanks!

Reply:
Hold on and will do more testing to confirm if the issue is fixed totally.